nanofi/examples/tail_file.c

```diff
@@ -0,0 +1,270 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+    *
+    *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+#include "api/nanofi.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <limits.h>
+#include <sys/stat.h>
+
+typedef struct flow_file_records {
+    flow_file_record ** records;
+    uint64_t len;
+} flow_file_records;
+
+typedef struct file_offset {
+    int offset;
+} file_offset;
+
+struct flow_file_records * flowfiles = NULL;
+struct file_offset * fileoffset = NULL;
+struct nifi_instance * instance = NULL;
+
+void transmit_flow_files(nifi_instance * instance) {
+    if (flowfiles) {
+        for (int i = 0; i < flowfiles->len; ++i) {
+            if (flowfiles->records[i]) {
+                transmit_flowfile(flowfiles->records[i], instance);
+            }
+        }
+    }
+}
+
+char * get_string(char * begin, char * end) {
```

Comment:
After I read the implementation I understood what this function does. 
Maybe _get_substring_ or _copy_substring_ as a name would be closer to what it does, but I hope you guys can come up with something even better. 
In the meanwhile a doxygen-style comment like in nanofi.h would be nice to describe the different usecases of the function (the string is expected to be null-terminated in case end is not specified).

I would also prefer to have const char * arguments given we don't modify the source. 
With the current signature the following wouldn't work:

```
const char * text = "nice code";
const char * text_part = get_string(text+5, NULL);
```

As this function is surely useful I think this should be moved to some of the nanofi core files (maybe a new one) to support future ourselves. 

For the cherry on top it could be trickily made variadic arg to allow the caller to only specify begin. :) (not mandatory at all)

Reply:
This function is gone from latest update.